.github/workflows/samples.yml

```diff
@@ -59,32 +59,37 @@ jobs:
       - name: "Testing Samples"
         run: ./build_tools/testing/test_samples.sh
 
-  # TODO(scotttodd): Convert web test scripts to use installed compiler packages and local runtime
-  #                  builds, rather than local compiler builds. The host tools are too slow to build
-  #                  on standard sized runners (and for casual users).
-
-  # web:
-  #   runs-on: ubuntu-20.04
-  #   # TODO: remove dependence on magic matching build dir names
-  #   env:
-  #     HOST_BUILD_DIR: build-host-install
-  #   steps:
-  #     - name: "Checking out repository"
-  #       uses: actions/checkout@v4.1.7
-  #       with:
-  #         submodules: true
-  #     - name: "Building host tools"
-  #       run: |
-  #         ./build_tools/github_actions/docker_run.sh  \
-  #           --env "CCACHE_NAMESPACE=gcr.io/iree-oss/base@sha256:dc314b4fe30fc1315742512891357bffed4d1b62ffcb46258b1e0761c737b446" \
-  #           gcr.io/iree-oss/base@sha256:dc314b4fe30fc1315742512891357bffed4d1b62ffcb46258b1e0761c737b446 \
-  #           ./build_tools/cmake/build_host_tools.sh "${HOST_BUILD_DIR}"
-  #     - name: "Testing Experimental Web Samples"
-  #       env:
-  #         IREE_EMPSCRIPTEN_BUILD_DIR: build-emscripten
-  #       run: |
-  #         ./build_tools/github_actions/docker_run.sh  \
-  #           --env "IREE_HOST_BUILD_DIR=${HOST_BUILD_DIR}" \
-  #           --env "IREE_EMPSCRIPTEN_BUILD_DIR=${IREE_EMPSCRIPTEN_BUILD_DIR}" \
-  #           gcr.io/iree-oss/emscripten@sha256:2dd4c52f1bb499ab365aad0111fe5538b685d88af38636b409b0cf6a576ab214 \
-  #           ./experimental/web/build_and_test_samples.sh
+  web:
+    runs-on: ubuntu-20.04
+    container: gcr.io/iree-oss/emscripten@sha256:2dd4c52f1bb499ab365aad0111fe5538b685d88af38636b409b0cf6a576ab214
+    env:
+      VENV_DIR: ${{ github.workspace }}/.venv
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4.1.7
+      - name: "Mark git safe.directory"
+        run: git config --global --add safe.directory '*'
```

Comment:
I am wondering if this is strictly needed to exec `build_tools/scripts/git/update_runtime_submodules.sh`.

Reply:
Error logs without it: https://github.com/ScottTodd/iree/actions/runs/10274398015/job/28431025243#step:4:14
```
Run ./build_tools/scripts/git/update_runtime_submodules.sh
  ./build_tools/scripts/git/update_runtime_submodules.sh
  shell: sh -e {0}
  env:
    VENV_DIR: /home/runner/work/iree/iree/.venv
+++ readlink -f -- ./build_tools/scripts/git/update_runtime_submodules.sh
++ dirname -- /__w/iree/iree/build_tools/scripts/git/update_runtime_submodules.sh
+ SCRIPT_DIR=/__w/iree/iree/build_tools/scripts/git
+ readarray -t RUNTIME_SUBMODULES
++ cat build_tools/scripts/git/runtime_submodules.txt
++ tr -d '\r'
+ git submodule sync
fatal: detected dubious ownership in repository at '/__w/iree/iree'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/iree/iree
Error: Process completed with exit code 1[2](https://github.com/ScottTodd/iree/actions/runs/10274398015/job/28431025243#step:4:2)8.
```

I copied this solution from https://github.com/iree-org/iree/blob/ea8b4fbe1364cf081758ad919e1107acac6befb4/build_tools/pkgci/build_linux_packages.sh#L104-L107 https://github.com/iree-org/iree/blob/ea8b4fbe1364cf081758ad919e1107acac6befb4/build_tools/python_deploy/build_linux_packages.sh#L100-L103

More info:
* https://stackoverflow.com/questions/72978485/git-submodule-update-failed-with-fatal-detected-dubious-ownership-in-reposit
* https://www.kenmuse.com/blog/avoiding-dubious-ownership-in-dev-containers/

This could be more tightly scoped than `"*"`, but it shouldn't matter much on an ephemeral GitHub-hosted runner using a container.